### DISCUSSION TITLE
How do you extract contextual information to help translators disambiguate?

### DISCUSSION THREAD

**Discussion by BrianGenisio** [i18next/i18next#2066] [Q&A]
Context of a string inside an application matters a lot when giving strings to translators to translate, especially when dealing with ambiguous language.

For example, I might have a string that says "Look a this table:" which could mean "a table of data" or "a kitchen table".  Ideally, my resource file in the source language would have some sort of contextual details along with it to pass to our translation system to help the translator properly translate.

How is the context in which the string used extracted in order to enable better translations?  Can arbitrary strings be added in our resource files that are ignored by i18next but used by our translation system?

**Comment by jamuhl** [comment]
You can call `i18next.t('key', 'defaultValue', 'context information')` or `i18next.t('key', { tDescription: 'context information' })` if your backend implementation supports passing context information to the backend (eg. locize.com's i18next-locize-backend https://github.com/locize/i18next-locize-backend )

The JSON format does not explicit support storing context information - but there is nothing that does not let you:

```js
{
  'key': 'value',
  'key#context': 'context for key'
}
```

Just avoid using reserved chars (`_`, `:`, `.`) to mark your keys as context. As long you do not call `i18next.t('key#context')` i18next will do nothing with that (just bloats the JSON)
